instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt

```diff
@@ -448,15 +448,16 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
         val options = MapboxRouteLineOptions.Builder(activity)
             .build()
         val routeLineApi = MapboxRouteLineApi(options)
+        val alternativeMetadata = mapboxNavigation.getAlternativeMetadataFor(routesUpdate)
         val result = routeLineApi.setNavigationRoutes(
             newRoutes = routesUpdate,
-            alternativeRoutesMetadata = mapboxNavigation.getAlternativeMetadataFor(routesUpdate)
+            alternativeRoutesMetadata = alternativeMetadata
         )
 
-        // the alternative route turns out to fully overlaps with the original route,
-        // hence the 1.0 offset
+        assertEquals(1, alternativeMetadata.size)
+        // the alternative route overlaps primary on ~92%
         assertEquals(
-            1.0,
+            0.9263153441670023,
             result.value!!.alternativeRouteLinesData[0].dynamicData.trimOffset!!.offset,
             0.0000000001
         )
```

Comment:
This test doesn't catch the original issue anymore. It was only reproducible because the routes were equal and we had to traverse till the last geometry point which made us run into the out-of-bounds problem.

I can help adjusting it.

Reply:
But this shouldn't be the case anymore. NN doesn't accept alternatives with geometry that equal to primary route

Reply:
Done.

Reply:
I pushed a new test to address this.